internal/backend/remote-state/s3/client_test.go

```diff
@@ -980,6 +981,156 @@ func TestS3LockingWritingHeaders(t *testing.T) {
 	}
 }
 
+func TestS3LockFileObjectTagging(t *testing.T) {
+	// Configured the aws config the same way it is done for the backend to ensure a similar setup as the actual main logic.
+	_, awsCfg, _ := awsbase.GetAwsConfig(context.Background(), &awsbase.Config{Region: "us-east-1", AccessKey: "test", SecretKey: "key"})
+	httpCl := &mockHttpClient{resp: &http.Response{StatusCode: http.StatusOK, Body: io.NopCloser(strings.NewReader(""))}}
+	s3Cl := s3.NewFromConfig(awsCfg, func(options *s3.Options) {
+		options.HTTPClient = httpCl
+	})
+
+	tests := []struct {
+		name         string
+		action       func(cl *RemoteClient) error
+		skipChecksum bool
+
+		lockTags map[string]string
+	}{
+		{
+			name:         "s3.Put with no tags",
+			skipChecksum: false,
+			action: func(cl *RemoteClient) error {
+				return cl.Put(t.Context(), []byte("test"))
```

Comment:
Most probably `cl.Put` is not what you want here, right?
Maybe `cl.Lock(...)` or `cl.s3Lock(...)`?